cedar-policy/src/api.rs

```diff
@@ -3384,6 +3384,54 @@ impl Context {
         cedar_policy_validator::context_schema_for_action(&schema.0, action.as_ref())
             .ok_or_else(|| ContextJsonError::missing_action(action.clone()))
     }
+
+    /// Combine two [`Context`]s, returning an error if they contain overlapping keys
+    pub fn merge(self, context: Self) -> Result<Self, ContextCreationError> {
```

Comment:
Sorry for being late. I was wondering if we can generalize the argument to `impl Iterator<SmolStr, ...>`, since `Context` implements the conversion.